MultiLang - German

FlorianReinprecht opened the pull request:
this PR should implement a simple language switcher to a german translation.

Additional languages can be added quite simple, actually. 

Thanks for creating this site!

FlorianReinprecht committed:
MultiLang - German

this PR should implement a simple language switcher to a german translation.

Additional languages can be added quite simple, actually. 

Thanks for creating this site!

SawyerHood commented on src/i18n.js:
```diff
@@ -0,0 +1,619 @@
+          "Jun - Oct": "Jun - Okt",
+          "Aug - Oct": "Aug - Okt",
+          "Oct - Apr": "Okt - Apr",
+          "Dec - Aug": "Dez - Aug",
```
This section is a little funky, can we use i18next's join functionality for this. It should greatly cut down on the number of translations

SawyerHood commented on src/i18n.js:
```diff
@@ -0,0 +1,619 @@
+          "11 PM - 8 AM": "23:00 - 08:00",
+          "4 AM - 5 PM": "04:00 - 17:00",
+          "5 PM - 4 AM": "17:00 - 04:00",
+          "4 AM - 8 AM & 5 PM - 7 PM": "04:00 - 08:00 & 17:00 - 19:00",
```
I feel like we should actually use moment to format these correctly rather than keep them in the translation table. Maybe we could move this to app state?

SawyerHood commented on src/i18n.js:
```diff
@@ -0,0 +1,619 @@
+          "Aug - Oct": "Aug - Oct",
+          "Oct - Apr": "Oct - Apr",
+          "Dec - Aug": "Dec - Aug",
+          "Sep - Dec": "Sep - Dec",
```
Same as the comment below we should probably use i18next's joining functionality https://www.i18next.com/translation-function/objects-and-arrays

SawyerHood commented on src/i18n.js:
```diff
@@ -0,0 +1,619 @@
+    resources: {
+      en: {
+        translations: {
+          "Not Yet Obtained": "Not Yet Obtained",
```
I'm not super familiar with i18next (I'm pretty familiar with internationalization as a whole though, I worked on FBT when I worked at Facebook in 2015 https://facebook.github.io/fbt/). If we don't have a translation in this table, will it just fallback to the key?

FlorianReinprecht replied:
Here you can find detailed information about the resolution of those keys. 
https://www.i18next.com/principles/translation-resolution#resolution-order

So basically, yes if the key is not found it will use either a specified fallback key or the key itself.

SawyerHood commented on src/App.tsx:
```diff
@@ -145,15 +154,62 @@ function Footer() {
+      <div className={text}>
+        {i18n.language === "en" ? "English" : "Deutsch"}
+      </div>
+      {i18n.language === "en" ? (
+        <English width="40px" height="40px" />
+      ) : (
+        <German width="40px" height="40px" />
+      )}
```
Can we remove the flags? I looks a little bit cleaner without them. Eventually I'll move all of this into a settings menu

SawyerHood reviewed (changes requested):
This is awesome! Thanks so much for doing this. My only immediate request is that we remove the flag svgs, they clash pretty hard with the rest of the site. It would be nice to use i18next's joining function for the date and time ranges, but if you don't feel up to that I can do it in a separate PR

FlorianReinprecht committed:
implement Moment.js localisation and language selection

Localisation of time and months was done with the help of  moment.js, as it was already in use. 

Instead of the flags a simple language selector was created.

FlorianReinprecht committed:
fix typo

FlorianReinprecht commented:
Hi,

thanks for the feedback. I tried to improve the localisation of time & dates with the help of moment.js, as it was already in use and you also suggested it als a possible solution. 

As for the flag problem: I changed the flags to a simple language picker for now. It should now look better on smartphones.

SawyerHood committed:
Merge branch 'master' into csb-1588020033044

SawyerHood commented:
Huge thanks for adding this @FlorianReinprecht! Closing this commit because I ended up merging it.
